CompanyApi/Controllers/CompanyController.cs

```diff
@@ -20,6 +23,75 @@ public ActionResult<Company> Create(CreateCompanyRequest request)
             return StatusCode(StatusCodes.Status201Created, companyCreated);
         }
 
+        [HttpGet]
+        public ActionResult<List<Company>> GetAllCompanies([FromQuery] int? pageSize, [FromQuery] int? pageIndex)
+        {
+            if (pageIndex == null || pageSize == null)
+            {
+                return StatusCode(StatusCodes.Status200OK, companies);
+            }
+
+            if ((pageIndex * pageSize) > companies.Count())
+            {
+                return StatusCode(StatusCodes.Status204NoContent);

```

Comment:
well: nice to use 204 for no found related data
suggestion: also return empty list to client